### ISSUE TITLE
v3:  `scope` not properly reflected in `(type|java)scriptreact`

### ISSUE THREAD

**Issue by rawhat** [lukas-reineke/indent-blankline.nvim#626]
In `v2` the scope updates when moving around between JSX elements.  In v3 with the configuration in the GIF, the scope does not update.  You can see that it does update correctly within the class/methods, so not sure if something is up with that case in particular.

The example below is for a super minimal config I tested with, but I have also confirmed in my full configuration.  I swear at one point I saw it actually work correctly with the same v3 setup, but I can't replicate that now.

![wezterm-gui_RX0IgqsP43](https://github.com/lukas-reineke/indent-blankline.nvim/assets/2095509/6c47a908-a9a3-45da-a89e-ca29dd6d493f)

Let me know if you need any more info!

Thank you :)

**Comment by lukas-reineke** [comment]
I think this is a bug in nvim-treesitter :thinking: 
It starts working when you make a change to the buffer.

ident-blankline just uses what is returned from nvim-treesitter `containing_scope`.
I will look into the nvim-treesitter code and see if I can figure it out.

https://github.com/lukas-reineke/indent-blankline.nvim/assets/12900252/c9a41c7e-6676-479f-9b10-21b88634eca4

**Comment by rawhat** [comment]
Hmm, interesting.  I'm not sure if `v2` is handling this differently somehow, but it seems to work without modification.  Also what you did above doesn't seem to fix it for me in my regular config, though it does in my stripped out config.  The setup is the same though.  I guess maybe something else I have is conflicting with that somehow.

**Comment by lukas-reineke** [comment]
> I'm not sure if v2 is handling this differently somehow

v2 by default was doing something different, but I don't want to support that anymore. nvim-treesitters scope is objectively a lot better, as long as it works :sweat_smile:.
I'll find some time next week to try to debug it.

**Comment by rawhat** [comment]
Totally understand.  Thank you so much for digging into this!

**Comment by lukas-reineke** [comment]
I looked more into this and discovered that there was a bunch of stuff broken.
Most importantly for this issue, nvim-treesitters caching was too aggressive. That's why it sometimes worked and sometimes didn't. Sometimes the cache matched the buffer, but sometimes it didn't.
I tried to fix the caching, but that caused way too many updates, which was too slow. I ended up changing how this works completely.
I pushed a new version that should work. And it should also be a lot faster.
